src/Symfony/Component/Process/Pipes/WindowsPipes.php

```diff
@@ -52,9 +52,11 @@ public function __construct($disableOutput, $input)
                 Process::STDERR => tempnam(sys_get_temp_dir(), 'err_sf_proc'),
             );
             foreach ($this->files as $offset => $file) {
-                if (false === $file || false === $this->fileHandles[$offset] = @fopen($file, 'rb')) {
+                if (false === $file || false === $h = @fopen($file, 'rb')) {
+                    $this->__destruct();
```

Comment:
I know this should work but to me it looks wrong to do it this way. Can't we rather move the code from `__destruct()` in its own method and reuse that then?